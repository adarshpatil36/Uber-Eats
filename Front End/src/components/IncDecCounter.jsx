import React from "react";

const IncDecCounter = ({ item, increaseQuantity, decreaseQuantity }) => {
  const increamentCount = () => {
    increaseQuantity(item);
  };
  const decreamentCount = () => {
    if (item.quantity !== 0) {
      decreaseQuantity(item);
    }
  };
  return (
    <span>
      <button className="quantity__minus" onClick={decreamentCount}>
        -
      </button>
      <label
        name="quantity"
        type="text"
        className="quantity__input"
        data-testid="incDecCounter"
      >
        {item.quantity || 0}
      </label>
      <button className="quantity__plus" onClick={increamentCount}>
        +
      </button>
    </span>
  );
};

export default IncDecCounter;
