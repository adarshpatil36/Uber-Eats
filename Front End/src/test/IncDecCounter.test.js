import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import ReactDOM from "react-dom";

import renderer from "react-test-renderer";
import IncDecCounter from "../components/IncDecCounter";

describe("IncDecCounter testing", () => {
  const item = {
    quantity: 20,
  };
  const increaseQuantity = () => {};
  const decreaseQuantity = () => {};

  it("renders IncDecCounter without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <IncDecCounter
        item={item}
        increaseQuantity={increaseQuantity}
        decreaseQuantity={decreaseQuantity}
      ></IncDecCounter>,
      div
    );
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <IncDecCounter
          item={item}
          increaseQuantity={increaseQuantity}
          decreaseQuantity={decreaseQuantity}
        ></IncDecCounter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
