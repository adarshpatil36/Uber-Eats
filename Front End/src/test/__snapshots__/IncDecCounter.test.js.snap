// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncDecCounter testing renders correctly 1`] = `
<span>
  <button
    className="quantity__minus"
    onClick={[Function]}
  >
    -
  </button>
  <label
    className="quantity__input"
    data-testid="incDecCounter"
    name="quantity"
    type="text"
  >
    20
  </label>
  <button
    className="quantity__plus"
    onClick={[Function]}
  >
    +
  </button>
</span>
`;
